source/extensions/filters/http/ext_authz/ext_authz.h

```diff
@@ -50,6 +50,35 @@ struct ExtAuthzFilterStats {
   ALL_EXT_AUTHZ_FILTER_STATS(GENERATE_COUNTER_STRUCT)
 };
 
+class ExtAuthzLoggingInfo : public Envoy::StreamInfo::FilterState::Object {
```

Comment:
For `ExtAuthzLoggingInfo` class here, could you just have filter_metadata_ which is only one introduced in this PR. But please also add implementation code in the ext_authz, e.g., set it from config etc.

Also corresponding unit test/integration test please :)  No need to have test consumer filter in this PR.